agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4433,49 +4433,48 @@ bool slave_thread::handle_channel_preference_query(Socket *sd, ieee1905_1::CmduM
 }
 
 /**
- * @brief check if channel is restricted in a given preference
+ * @brief Get the channel preference 
  * 
  * @param channel channel to check
- * @param preference channel preference
- * @return true if channel is restricted
- * @return false if channel is not restricted
+ * @param preference preference 
+ * @return NON_OPERABLE if channel is restricted, channel preference otherwise
  */
-static bool is_channel_restricted(const beerocks::message::sWifiChannel channel,
-                                  const son::wireless_utils::sChannelPreference &preference)
+static wfa_map::cPreferenceOperatingClasses::ePreference
+get_channel_preference(const beerocks::message::sWifiChannel channel,
+                       const son::wireless_utils::sChannelPreference &preference)
 {
     uint8_t center_channel = 0;
     auto bw                = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
     auto operating_class   = wireless_utils::get_operating_class_by_channel(channel);
 
+    // consider channels in different operating classes as restricted
+    if (operating_class != preference.oper_class) {
+        return wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
+    }
     // operating classes 128,129,130 use center channel **unlike the other classes**,
     // so convert channel and bandwidth to center channel.
     // For more info, refer to Table E-4 in the 802.11 specification.
     if (operating_class == 128 || operating_class == 129 || operating_class == 130) {
         center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
     }
 
-    // restrict channels from other operating classes
-    if (operating_class != preference.oper_class) {
-        return true;
-    }
     // According to Table 23 in the MultiAP Specification, an empty channel list field
     // indicates that the indicated preference applies to all channels in the operating class.
     if (preference.channels.empty()) {
-        return true;
+        return wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
     }
     // explicitely restrict non-operable channels
+    auto channel_to_check =
+        (operating_class == 128 || operating_class == 129 || operating_class == 130)
+            ? center_channel
+            : channel.channel;
     for (const auto &ch : preference.channels) {
-        if (operating_class == 128 || operating_class == 129 || operating_class == 130) {
-            if (center_channel == ch.channel) {
-                return true;
-            }
-        } else {
-            if (channel.channel == ch.channel) {
-                return true;
-            }
+        if (channel_to_check == ch.channel) {
+            wfa_map::cPreferenceOperatingClasses::ePreference::NON_OPERABLE;
```

Comment:
The `return` seems to have gone missing here :-)

But again, I meant `return preference.preference;`